openapi-cli/src/main/java/io/ballerina/openapi/cmd/OASContractGenerator.java

```diff
@@ -124,7 +124,16 @@ public void generateOAS3DefinitionsAllService(Path servicePath, Path outPath, St
             List<String> fileNames = new ArrayList<>();
             for (OASResult definition : openAPIDefinitions) {
                 try {
+                    List<OpenAPIMapperDiagnostic> definitionDiagnostics = definition.getDiagnostics();
+                    boolean hasErrors = definitionDiagnostics.stream()
+                            .anyMatch(d -> DiagnosticSeverity.ERROR.equals(d.getDiagnosticSeverity()));
                     this.diagnostics.addAll(definition.getDiagnostics());
+                    if (hasErrors) {
+                        //This is only stop code generation for the given ERROR severity.
```

Comment:
For me, this comment look redundant
```suggestion
                        //This is only stop code generation for the given ERROR severity.
```